File: backend/src/main/java/com/petcare/model/PetInfo.java
package com.petcare.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "pet_info")
@Data
public class PetInfo {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String breed;
    
    @Column(name = "date_of_birth", nullable = false)
    private LocalDate dateOfBirth;
    
    private String gender;
    
    private String color;
    
    @Column(name = "microchip_id")
    private String microchipId;
    
    @Column(name = "image_path")
    private String imagePath;
    
    @Column(columnDefinition = "TEXT")
    private String notes;
}

File: backend-python/app/models/pet_info.py
from sqlalchemy import Column, Integer, String, Date, Text
from app.database import Base

class PetInfo(Base):
    __tablename__ = "pet_info"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    breed = Column(String, nullable=False)
    date_of_birth = Column(Date, nullable=False)
    gender = Column(String)
    color = Column(String)
    microchip_id = Column(String)
    image_path = Column(String)
    notes = Column(Text)
